13-Tenant Security Solution Setup/Scripts/RemediationScripts/Remediate-ConfigASCTier.ps1

```diff
@@ -0,0 +1,389 @@
+    )
+
+    Write-Host "======================================================"
+    Write-Host "Starting to remediate config ASC tier for subscription [$($SubscriptionId)]..."
+    Write-Host "------------------------------------------------------"
+
+    if($PerformPreReqCheck)
+    {
+        try 
+        {
+            Write-Host "Checking for pre-requisites..."
+            Pre_requisites
+            Write-Host "------------------------------------------------------"     
+        }
+        catch 
+        {
+            Write-Host "Error occured while checking pre-requisites. ErrorMessage [$($_)]" -ForegroundColor Red    
+            break
+        }
+    }
+
+    # Connect to AzAccount
+    $isContextSet = Get-AzContext
+    if ([string]::IsNullOrEmpty($isContextSet))
+    {       
+        Write-Host "Connecting to AzAccount..."
+        Connect-AzAccount
+        Write-Host "Connected to AzAccount" -ForegroundColor Green
+    }
+
+    # Setting context for current subscription.
+    $currentSub = Set-AzContext -SubscriptionId $SubscriptionId
+
+    Write-Host "Metadata Details: `n SubscriptionId: [$($SubscriptionId)] `n AccountName: [$($currentSub.Account.Id)] `n AccountType: [$($currentSub.Account.Type)]"
+    Write-Host "------------------------------------------------------"
+    Write-Host "Starting with Subscription [$($SubscriptionId)]..."
+
+
+    Write-Host "Step 1 of 3: Validating whether the current user [$($currentSub.Account.Id)] have the required permissions to run the script for Subscription [$($SubscriptionId)]..."
+
+    # Safe Check: Checking whether the current account is of type User
+    if($currentSub.Account.Type -ne "User")
+    {
+        Write-Host "Warning: This script can only be run by user account type." -ForegroundColor Yellow
+        break;
+    }
+
+    # Safe Check: Current user need to be either Contributor or Owner for the subscription
+    $currentLoginRoleAssignments = Get-AzRoleAssignment -SignInName $currentSub.Account.Id -Scope "/subscriptions/$($SubscriptionId)";
+
+    if(($currentLoginRoleAssignments | Where { $_.RoleDefinitionName -eq "Owner"  -or $_.RoleDefinitionName -eq 'Contributor' } | Measure-Object).Count -le 0)
+    {
+        Write-Host "Warning: This script can only be run by an Owner or Contributor of subscription [$($SubscriptionId)] " -ForegroundColor Yellow
+        break;
+    }
+
+    # Declaring required ASC type and pricing tier
+    $reqASCTierResourceTypes = "VirtualMachines","SqlServers","AppServices","StorageAccounts","KubernetesService","ContainerRegistry","KeyVaults","SqlServerVirtualMachines";		
+    $reqASCTier = "Standard";
+    $reqProviderName = "Microsoft.Security"
+    $isProviderRegister = $true
+
+    # Checking IsProviderRegister with 'Microsoft.Security' provider
+    $registeredProvider =  Get-AzResourceProvider -ListAvailable | Where-Object { $_.RegistrationState -eq "Registered" -and $_.ProviderNamespace -eq $reqProviderName }
```

Comment:
Check if we can check for specific provider Get-AzResourceProvider -ProviderNamespace "" instead of listing all providers and then filtering for ASC.

Reply:
Added parameter to do filtering using provider name